code/modules/mob/living/carbon/human/life.dm

```diff
@@ -327,4 +331,39 @@
 		adjustOxyLoss(5)
 		adjustBruteLoss(1)
 
+/mob/living/carbon/human/proc/handle_affinity()
+	if(check_affinity(mind, AFFINITY_EARTH))

```

Comment:
I'm doing it this way because witches can only have a single affinity at a time. Datums for affinities could result in someone being attuned to both water and fire, something I want to avoid.